src/odemis/driver/ephemeron.py

```diff
@@ -0,0 +1,747 @@
+        :param url (str): The url address to use with the OPC UA protocol
+        :param namespace (str): The object name identifier to use for the OPC UA protocol
+        """
+        super().__init__(name, role, **kwargs)
+
+        self._channel = channel
+        self._url = url
+        self._namespace = namespace
+
+        # server_sim is only used for simulation of the opcServer if it is non-existent
+        self._opc_server_sim: Optional[MightyEBICSimulator] = None
+        self._opc_client: Optional[UaClient] = None
+        self._error_msg: Optional[str] = None
+        self._t_simserver: Optional[threading.Thread] = None
+        # hardcoded now, eventually -> get resolution from the controller/detector
+        self._ebeam_res = ((1, 1), (4096, 4096))
+        self._server_exception: Optional[Exception] = None
+
+        if "localhost" in self._url:
+            # if the device should be simulated, start a simulated server first
+            self._opc_server_sim = MightyEBICSimulator(self._url, self._namespace, self)
+            self._t_simserver = threading.Thread(target=self._start_opc_simserver)
+            self._t_simserver.start()
+            # start the simulated server threaded but wait for it to be ready (running)
+            while not self._opc_server_sim.ready:
+                time.sleep(0.1)
+                if self._server_exception:
+                    raise HwError(self._server_exception.args[0])
+
+        # fire up the client needed for the opcua communication with the server
+        self._opc_client = UaClient(self._url, DEFAULT_TIMEOUT)
+        asyncio.run(self._opc_client.initialize_client())  # this will wait/block until the client is initialized
+
+        self.numberOfChannels = model.IntEnumerated(1,
+                                                    set(range(1, MAX_NUMBER_OF_CHANNELS + 1)),
+                                                    setter=self.on_chan_num_change)
+        # Oversampling is a digital low-pass filter that removes high-frequency
+        # noise and sends back a single clean 18-bit value.
+        self.oversampling = model.IntEnumerated(0, {0, 2, 4, 8, 16, 32, 64})
+        self.oversampling.subscribe(self.on_oversampling_change)
+        self.resolution = model.ResolutionVA(self._ebeam_res[1], (self._ebeam_res[0], self._ebeam_res[1]), unit="px")
+        self.dwellTime = model.FloatContinuous(1e-5, (0.1e-6, 1000), unit="s")  # 10 us default
+
+        self.data = EBICDataFlow(self)
+        self._acquisition_thread: Optional[threading.Thread] = None
+        self._acquisition_lock = threading.Lock()
+        self._acquisition_must_stop = threading.Event()
+
+        # Special event to request software unblocking on the scan
+        self.softwareTrigger = model.Event()
+
+        self._metadata[model.MD_DET_TYPE] = model.MD_DT_NORMAL
+        self._metadata[model.MD_HW_VERSION] = self._hwVersion
+        self._metadata[model.MD_SW_VERSION] = self._swVersion
```

Comment:
if the versions are unknown, no need to store them in the metadata. 
Once you can read the version number from a OPC node, then you can add this (back).